src/codec/decoder/audio.rs

```diff
@@ -130,3 +138,38 @@ impl AsMut<Context> for Audio {
 		&mut self.0
 	}
 }
+
+pub struct AudioFrameIter<'a, 'b> {
+	audio: &'a mut Audio,
+	packet: &'b mut packet::Packet,
+}
+
+impl<'a, 'b> Iterator for AudioFrameIter<'a, 'b> {
+    type Item = Result<Option<frame::Audio>, Error>;
+
+    fn next(&mut self) -> Option<Result<Option<frame::Audio>, Error>> {
+		unsafe {
+			if !self.packet.is_empty() {
+				let mut out = frame::Audio::empty();
+				let mut got: c_int = 0;
+				let packet = self.packet.as_mut_ptr();
+
+				match avcodec_decode_audio4(self.audio.as_mut_ptr(), out.as_mut_ptr(), &mut got, packet) {
+					e if e < 0 => Some(Err(Error::from(e))),
+					n => {
+						(*packet).data = (*packet).data.offset(n as isize);
+						(*packet).size -= n;
```

Comment:
I think it would be better if the packet was borrowed immutably, and then copied over, then these two attributes modified in place in the internal packet.